gear/gear/database.py

```diff
@@ -77,6 +80,19 @@ async def aexit(acontext_manager, exc_type=None, exc_val=None, exc_tb=None):
     return await acontext_manager.__aexit__(exc_type, exc_val, exc_tb)
 
 
+async def resolve_test_db_endpoint(sql_config: SQLConfig) -> SQLConfig:
+    service_name, namespace = sql_config.host[: -len('.svc.cluster.local')].split('.', maxsplit=1)
+    await kubernetes_asyncio.config.load_kube_config()
+    async with kubernetes_asyncio.client.ApiClient() as api:
+        client = kubernetes_asyncio.client.CoreV1Api(api)
+        db_service = await client.read_namespaced_service(service_name, namespace)
+        db_pod = await client.read_namespaced_pod(f'{db_service.spec.selector["app"]}-0', namespace)
+        sql_config_dict = sql_config.to_dict()
+        sql_config_dict['host'] = db_pod.status.host_ip
```

Comment:
this feels a bit flaky no? Are we not able to access this through the service's DNS name?

Reply:
Ah. The workers. And exposing the service to the workers would require a private load balancer, etc. etc. annoying costly.

Reply:
ya exactly. I didn't want to allocate a whole internal load balancer just to run a few couple minute migrations.

Reply:
It does feel a bit flaky, which is why I specifically don't do this if you're running in k8s. I am implicitly assuming here that the DB pod is not going to be relocated while the create_database job is running, which I guess would pose a problem anyway. This seemed the most reasonable, albeit a little janky, approach in terms of cost and complexity given that these are just test databases.